src/Symfony/Component/Form/Util/ParamFilesMerger.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Util;
+
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+
+/**
+ * @author Priyadi Iman Nurcahyo <priyadi@rekalogika.com>
+ */
+class ParamFilesMerger
+{
+    private array $path;
+    private array $params;
+    private array $files;
+
+    /**
+     * @param array $path   The path to the current element, empty means the root
+     * @param array $params The parameters
+     * @param array $files  The files
+     */
+    public function __construct(array $path, array $params, array $files)
+    {
+        $this->path = $path;
+        $this->params = $params;
+        $this->files = $files;
+    }
+
+    public function getResult()
+    {
+        $paramsValue = $this->getParamsValue();
+        $filesValue = $this->getFilesValue();
+
+        if (null === $paramsValue) {
+            return $filesValue;
+        } elseif (\is_array($paramsValue)) {
```

Comment:
all `else` statements can be removed to improve readability

Reply:
Tried that, but considering the nature of the bugs we are having here in the past 6 months or so, I don't think it would improve readability. I suggest leaving it as is.

Reply:
I'd also prefer removing the elseif/else when possible
we have tests to spot bugs